File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        dec, inc, ans = [1] * 3
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                inc += 1
                dec = 1
            elif nums[i] < nums[i - 1]:
                dec += 1
                inc = 1
            else:
                dec = inc = 1
            ans = max(ans, dec, inc)
        return ans 

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int cntI = 1;
        int cntD = 1;
        int ans = 1 ;
        for(int i = 1 ; i < nums.length ; i ++){
            if(nums[i] > nums[i - 1]) cntI ++ ;
            else cntI = 1 ;
            if(nums[i] < nums[i - 1]) cntD ++ ;
            else cntD = 1 ;
            ans = Math.max(ans, Math.max(cntI, cntD));
        }
        return ans ;
    }
}